apps/TideForker/main.cpp

```diff
@@ -63,14 +63,25 @@ int main(int argc, char* argv[])
     COMMAND_LINE_PARSER_CHECK(CommandLineHelper, "tideForker");
 
     {
-        MPIChannelPtr worldChannel(new MPIChannel(argc, argv));
-        const int rank = worldChannel->getRank();
-        MPIChannelPtr localChannel(new MPIChannel(*worldChannel, 0, rank));
-        MPIChannelPtr localChannel2(new MPIChannel(*worldChannel, 0, rank));
-        MPIChannelPtr mainChannel(new MPIChannel(*worldChannel, 0, rank));
+        auto worldComm = MPICommunicator{argc, argv};
```

Comment:
why not ```MPICommunicator worldComm{argc, argv};```

Reply:
Mostly a style question, inspired from here (see points 4 and 5): https://herbsutter.com/2013/08/12/gotw-94-solution-aaa-style-almost-always-auto/